src/Schema/AbstractAsset.php

```diff
@@ -51,10 +51,10 @@ protected function _setName($name)
             $name          = $this->trimQuotes($name);
         }
 
-        if (strpos($name, '.') !== false) {
-            $parts            = explode('.', $name);
-            $this->_namespace = $parts[0];
-            $name             = $parts[1];
+        $lastDotPos = strrpos($name, '.');
+        if ($lastDotPos !== false) {
+            $this->_namespace = substr($name, 0, $lastDotPos);
+            $name             = substr($name, $lastDotPos + 1);
```

Comment:
@morozov would you be fine with PR which will throw if more than 2 parts are present? Currently, such case is silently ignored leading to a corrupted parse result.